pkg/services/publicdashboards/api/middleware.go

```diff
@@ -41,7 +55,7 @@ func RequiresValidAccessToken(publicDashboardService publicdashboards.Service) f
 		}
 
 		if !exists {
-			c.JsonApiErr(http.StatusBadRequest, "Invalid access token", nil)
+			c.JsonApiErr(http.StatusNotFound, "Invalid access token", nil)
```

Comment:
Shouldn't we change this error message too?

Reply:
Yep